modules/utils/include/hephaestus/utils/file.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+#include <filesystem>
+#include <span>
+
+namespace heph::utils {
+
+[[nodiscard]] auto readFile(const std::filesystem::path& path) -> std::string;
+[[nodiscard]] auto readBinaryFile(const std::filesystem::path& path) -> std::vector<std::byte>;
+
+void writeStringToFile(const std::filesystem::path& path, std::string_view content);
+void writeBufferToFile(const std::filesystem::path& path, std::span<std::byte> content);
+
+/// This class allows to create a file that is removed when the class goes out of scope (RAII).
+/// This is very useful in tests to avoid having dangling files.
+class SelfDestructingPath {
```

Comment:
```suggestion
class TemporaryFile {
```

Reply:
The reason why I didn't go for `TemporaryFile` is that in linux terms a temp file is created in `/tmp` which is emptied only on bootstrap. Here I wanted to make clear that the file is destroyed when you go out of scope.

But if you think `TemporaryFile` is more clear I will update it

Reply:
I like `TemporaryFile` tbh.

@floriantschopp Naming discussion, do you have an opinion?

Reply:
Agree with @filippobrizzi here, for a temporary file I would have assumed it to be in `/tmp` and handleded as Filippo described.

Reply:
How about volatile?

Reply:
What about `ScopedPath`? like `std::scoped_lock`?